### DISCUSSION TITLE
Is there an alternative to the cart functions of v2 and before?

### DISCUSSION THREAD

**Discussion by nfinzer** [etsy/open-api#989] [Q&A] [answered]
My client has a web app that allows customers to preview and customize the products she sells on etsy. The code used to create a guest cart and allow the user to add the customized product to their etsy cart from the website. I see cart and guest card were deprecated. Is there another means of accomplishing this with v3? I've converted all the listing api calls to using OAuth2 tokens and the other v3 changes and it is working well, but I haven't been able to find anything about a workaround for this. I don't know why etsy would make it harder to buy things. We have a workaround but it is definitely a subpar user experience. 

Thanks in advance.

**Comment by Gareth064** [comment, marked as answer]
Hey. There is nothing in V3 which replicates this functionality I'm afraid.

I believe Etsy are (now) firmly against any 3rd-party service which tries to conduct Sales outside of the main Etsy App/Website, even if it is still going through the Etsy Cart etc. They have refused App approvals in the past because they were suspected of doing such a practice.

They probably want all Buyer interaction to happen through their own platform and not someone else's, so they can farm as much data about how the Buyer is using the platform etc.

**Comment by nfinzer** [comment]
Thanks for your reply. Etsy ought to reconsider this decision.

**Reply by Gareth064** [reply]
If you want to get a statement from them yourself then drop them an email at developer@etsy.com and see what their current and future position is on this.

If they give you a response, then let us all know back here on this discussion 👍🏻

**Reply by nfinzer** [reply]
Thanks, I have emailed them. I'll update if I get any response.
